src/geo/map/cartodb-layer.js

```diff
@@ -100,6 +101,58 @@ var CartoDBLayer = LayerModelBase.extend({
     var meta = this.get('meta');
     var stats = meta && meta.stats;
     return stats && stats.estimatedFeatureCount;
+  },
+
+  getSourceId: function () {
+    var source = this.getSource();
+    return source && source.id;
+  },
+
+  getSource: function () {
+    return this.get('source');
+  },
+
+  setSource: function (source, options) {
+    this.set('source', source, options);
```

Comment:
@IagoLast since this will be an API method... do you think we should check that `source` is a valid source here?

Reply:
I believe in order to make this method bakwards compatible (BC) it should accept both (string-id and AnalysisNode).

Once builder gets fixed we can remove the BC changes and add a check here

Reply:
This is still not being used externally so there's no need to make it backwards compatible IMO.